### PR TITLE
Initial support for kind

### PR DESCRIPTION
**Pull request by BenPope**
Currently based on #16 as that's a dependency.

The script currently only works with Linux and only tested with Ubuntu.

Useful for Mac: https://www.thehumblelab.com/kind-and-metallb-on-mac/

Signed-off-by: Ben Pope <ben@vectorized.io>

### CODE REVIEW

File: kind/make.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  --set configInline.address-pools[0].addresses[0]="$ADDRESSES"
+
+#Install redpanda
+helm install \
+ --create-namespace \
+ --namespace redpanda \
+ redpanda ../redpanda \
+ --set loadBalancer.enabled=true \
+ --set image.repository=localhost:5000/redpanda \
+ --set image.tag=b2783df
```
Comment (BenPope) [inline kind/make.sh:51]: oops

File: kind/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Testing a rolling restart:
+
+```sh
+kubectl -n redpanda rollout restart statefulset redpanda
+```
+
+### Accessing Kubernetes
+
+I'm not a fan of [kubernetes-dashboard](https://github.com/kubernetes/dashboard#kubernetes-dashboard) and it's a pain to set up.
```
Comment (RafalKorepta) [inline kind/README.md:48]: Maybe project [octant](https://octant.dev/) can be listed hear too.

File: kind/make.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+echo ""
+kubectl -n redpanda rollout status -w statefulset/redpanda
+echo "
+Remove cluster with:
+
+  $ kind delete cluster --name redpanda
+
+Add these to /etc/hosts:
+"
+kubectl -n redpanda get svc | grep redpanda- | awk '{printf "%-15s %s %s.redpanda.redpanda.svc.cluster.local\n", $4, $1, $1}' || true
```
Comment (RafalKorepta) [inline kind/make.sh:62]:
```suggestion
kubectl -n redpanda get svc redpanda-bootstrap | awk '{printf "%-15s %s %s.redpanda.redpanda.svc.cluster.local\n", $4, $1, $1}' || true
```

File: kind/make.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  --namespace metallb-system \
+  metallb bitnami/metallb \
+  -f metallb-values.yaml \
+  --set configInline.address-pools[0].addresses[0]="$ADDRESSES"
+
+#Install redpanda
+helm install \
+ --create-namespace \
+ --namespace redpanda \
+ redpanda ../redpanda \
```
Comment (RafalKorepta) [inline kind/make.sh:48]: What do you think about extract `redpanda` to local variable?
All places that uses the release name could use mentioned local variable.

File: kind/make.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+docker pull vectorized/redpanda:latest
+docker tag vectorized/redpanda:latest localhost:5000/redpanda:latest
+docker push localhost:5000/redpanda:latest
+
+NODES=$(kubectl get nodes -o json | jq -r '.items[].status.addresses | select(.[].address | startswith("redpanda-worker")) | .[] | select(.type == "InternalIP").address')
+SUBNET=$(echo "$NODES" | head -n1 | cut -d. -f 1,2).255
+ADDRESSES="$SUBNET.1-$SUBNET.254"
+
+# Install metallb
+helm repo add bitnami https://charts.bitnami.com/bitnami
```
Comment (RafalKorepta) [inline kind/make.sh:36]: What do you think to create empty helm chart that has `requirments.yaml` pointing to `metallb` and `redpanda`?

Comment (BenPope) [reply kind/make.sh:36]: Sounds neat.

File: kind/make.sh (hunk 4 above)
Comment (BenPope) [reply kind/make.sh:48]: This script certainly isn't production ready!

File: kind/make.sh (hunk 3 above)
Comment (BenPope) [reply kind/make.sh:62]: Ahh. This kind of relies on #16 - so it's supposed to grab the LoadBalance services. But yeah, the grep is wrong these days. I can harden this.

File: kind/README.md (hunk 2 above)
Comment (BenPope) [reply kind/README.md:48]: These types of opinions probably should be in this repo. I'll remove it.

File: kind/make.sh (hunk 3 above)
Comment (RafalKorepta) [reply kind/make.sh:62]: I see. Thanks for explanation

Comment (BenPope) [reply kind/make.sh]: Rebased on #16 and hardened it with some jq magic.

**Pull request closed without merging**